code/modules/research/bepis.dm

```diff
@@ -0,0 +1,225 @@
+	//Stock part variables
+	var/power_saver = 1
+	var/inaccuracy_percentage = 1.5
+	var/positive_cash_offset = 0
+	var/negative_cash_offset = 0
+
+/obj/machinery/rnd/bepis/attackby(obj/item/O, mob/user, params)
+	if(default_deconstruction_screwdriver(user, "chamber_open", "chamber", O))
+		update_icon_state()
+		return
+	else if(default_deconstruction_crowbar(O))
+		return
+	if(!is_operational())
+		to_chat(user, "<span class='notice'>[src] can't accept money when it's not functioning.</span>")
+		return
+	if(istype(O, /obj/item/holochip) || istype(O, /obj/item/stack/spacecash))
+		var/deposit_value = O.get_item_credit_value()
+		banked_cash += deposit_value
+		qdel(O)
+		say("Deposited [deposit_value] credits into storage.")
+		update_icon_state()
+		return
+	return ..()
+
+/obj/machinery/rnd/bepis/RefreshParts()
+	var/C = 0
+	var/M = 0
+	var/L = 0
+	var/S = 0
+	for(var/obj/item/stock_parts/capacitor/Cap in component_parts)
+		C += ((Cap.rating - 1) * 0.1)
+	power_saver = 1 - C
+	for(var/obj/item/stock_parts/manipulator/Manip in component_parts)
+		M += ((Manip.rating - 1) * 250)
+	positive_cash_offset = M
+	for(var/obj/item/stock_parts/micro_laser/Laser in component_parts)
+		L += ((Laser.rating - 1) * 250)
+	negative_cash_offset = L
+	for(var/obj/item/stock_parts/scanning_module/Scan in component_parts)
+		S += ((Scan.rating - 1) * 0.25)
+	inaccuracy_percentage = (1.5 - S)
+
+/obj/machinery/rnd/bepis/proc/depositcash()
+	var/deposit_value = 0
+	deposit_value = banking_amount
+	if(deposit_value == 0)
+		update_icon_state()
+		say("Attempting to deposit 0 credits. Aborting.")
+		return
+	deposit_value = CLAMP(round(deposit_value, 1), 1, 30000)
+	if(!account)
+		say("Cannot find user account. Aborting.")
+		return
+	if(!account.has_money(deposit_value))
+		say("You do not possess enough credits.")
+		return
+	else
+		account.adjust_money(-deposit_value) //The money vanishes, not paid to any accounts.
+		banked_cash += deposit_value
+		use_power(1000 * power_saver)
+		say("Cash deposit successful. There is [banked_cash] in the chamber.")
+	update_icon_state()
+	return
+
+/obj/machinery/rnd/bepis/proc/calcsuccess()
+	var/turf/dropturf = get_turf(pick(view(1,src)))
+	var/gauss_major = 0
+	var/gauss_minor = 0
+	var/gauss_real = 0
+	if(!dropturf) //Check to verify the turf exists and the reward isn't lost somehow.
+		dropturf = drop_location()
+	gauss_major = (gaussian(major_threshold, std) - negative_cash_offset)	//This is the randomized profit value that this experiment has to surpass to unlock a tech.
+	gauss_minor = (gaussian(minor_threshold, std) - negative_cash_offset)	//And this is the threshold to instead get a minor prize.
+	gauss_real = (gaussian(banked_cash, std*inaccuracy_percentage) + positive_cash_offset)	//this is the randomized profit value that your experiment expects to give.
+	say("Real: [gauss_real]. Minor: [gauss_minor]. Major: [gauss_major].")
+	if(gauss_real >= gauss_major) //Major Success.
```

Comment:
banked_cash will always be set to 0 after this, do it before the ifs. Same for flick and icon state.